### PR TITLE
feat: Apply character relic sets to the relic tab filter bar

### PR DESCRIPTION
**Pull request by WolfBungalow**
# Pull Request

## Description

* Added 'Select relics' button to RelicsTab. This button filters relics based on the Recommended Character's relic and ornament sets as defined in the Character Optimizer tab.

## Related Issue
* https://github.com/fribbels/hsr-optimizer/issues/476

## Checklist

- [X] I have added commit messages that are descriptive and meaningful.
- [X] I have tested the changes locally.
- [X] I have reviewed the code changes.

## Screenshots
image

### CODE REVIEW

File: src/lib/tabs/tabRelics/RelicFilterBar.tsx [hunk 1]
```diff
@@ -239,6 +239,27 @@ export default function RelicFilterBar(props: {
     window.store.getState().setScoringModalOpen(true)
   }
 
+  function selectRecommendationCharacterRelics() {
+    if (!currentlySelectedCharacterId) return
+  
+    const characterPreferredSets = window.store.getState().characters.find(
+      (character) => character.id === currentlySelectedCharacterId,
+    )
```
Comment (FI00ds) [inline src/lib/tabs/tabRelics/RelicFilterBar.tsx:247]: maybe `characterPreferredSets = window.store.getState().getCharacterById(character.id)` would be more concise here?

Comment (WolfBungalow) [reply src/lib/tabs/tabRelics/RelicFilterBar.tsx:247]: I'm always up for better naming.

Comment (WolfBungalow) [reply src/lib/tabs/tabRelics/RelicFilterBar.tsx]: const characterPreferredSets = window.store.getState().charactersById[currentlySelectedCharacterId]

**Comment by WolfBungalow** [comment]
Implemented @FI00ds change.

**Comment by fribbels** [comment]
I'm not sure I agree with pulling relics based on the optimizer page, most of the characters are just going to be empty sets since people tend to not choose set filters - This kinda overlaps with: https://github.com/fribbels/hsr-optimizer/issues/247

The ideal workflow should be that we predefine a set of [high / med / low] value relics sets per character and let the user choose a preset that way. The difficulty is just creating and maintaining that relic mapping across 60 characters x 40 sets

**Pull request closed without merging**